app/routers/weather_forecast.py

```diff
@@ -0,0 +1,232 @@
+import datetime
+from dotenv import load_dotenv
+import os
+import requests
+
+
+# This feature requires an API KEY - get yours free @ visual-crossing-weather.p.rapidapi.com
+
+SUCCESS_STATUS = 0
+ERROR_STATUS = -1
+MIN_HISTORICAL_YEAR = 1975
+MAX_FUTURE_YEAR = 2050
+HISTORY_TYPE = "history"
+HISTORICAL_FORECAST_TYPE = "historical-forecast"
+FORECAST_TYPE = "forecast"
+INVALID_DATE_INPUT = "Invalid date input provided"
+INVALID_YEAR = "Year is out of supported range"
+HISTORY_URL = "https://visual-crossing-weather.p.rapidapi.com/history"
+FORECAST_URL = "https://visual-crossing-weather.p.rapidapi.com/forecast"
+HEADERS = {'x-rapidapi-host': "visual-crossing-weather.p.rapidapi.com"}
+BASE_QUERY_STRING = {"aggregateHours": "24", "unitGroup": "metric", "dayStartTime": "00:00:01",
+                     "contentType": "json", "dayEndTime": "23:59:59", "shortColumnNames": "True"}
+HISTORICAL_AVERAGE_NUM_OF_YEARS = 3
+NO_API_RESPONSE = "No response from server"
+
+
+def validate_date_input(requested_date):
+    """ date validation.
+    Args:
+        requested_date (date) - date requested for forecast.
+    Returns:
+        (bool) - validate ended in success or not.
+        (str) - error message.
+    """
+    if isinstance(requested_date, datetime.date):
+        if MIN_HISTORICAL_YEAR <= requested_date.year <= MAX_FUTURE_YEAR:
+            return True, None
+        else:
+            return False, INVALID_YEAR
+    else:
+        return False, INVALID_DATE_INPUT
+
+
+def get_data_from_weather_api(url, input_query_string):
```

Comment:
We should cache it somehow to prevent long loading times.

Bonus: it might be better to use async/await here, if possible.

Reply:
I've added cache support :-)